interop-testing/src/test/java/io/grpc/testing/integration/NettyFlowControlTest.java

```diff
@@ -0,0 +1,315 @@
+import io.netty.channel.socket.nio.NioSocketChannel;
+import io.netty.handler.codec.http2.DefaultHttp2LocalFlowController;
+import io.netty.handler.codec.http2.Http2LocalFlowController;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.SocketAddress;
+import java.net.SocketOption;
+import java.net.UnknownHostException;
+import java.nio.channels.ServerSocketChannel;
+import java.nio.channels.SocketChannel;
+import java.nio.channels.spi.SelectorProvider;
+import java.text.MessageFormat;
+import java.time.Clock;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.logging.Formatter;
+import java.util.logging.Handler;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+
+@RunWith(JUnit4.class)
+public class NettyFlowControlTest {
+
+  // in bytes
+  private static final int LOW_BAND = 1024 * 1024;
+  private static final int MED_BAND = 2 * 1024 * 1024;
+  private static final int HIGH_BAND = 5 * 1024 * 1024;
+
+  // in milliseconds
+  private static final int LOW_LAT = 50;
+  private static final int MED_LAT = 100;
+  private static final int HIGH_LAT = 200;
+
+  // in bytes
+  private static final int TINY_WINDOW = 1;
+  private static final int REGULAR_WINDOW = 64 * 1024;
+  private static final int BIG_WINDOW = 1024 * 1024;
+  private static final int MAX_WINDOW = 8 * 1024 * 1024;
+
+  private static Logger logger = Logger.getLogger("io.grpc.netty.NettyClientHandler");
+  private static Handler logHandler;
+  private static ManagedChannel channel;
+  private static Server server;
+  private static String localhost = "127.0.0.1";
+  private static TrafficControlProxy proxy;
+
+  // TODO: make ports/proxy arguments
+  private final int port = 5001;
+  private final int proxyPort = 5050;
+  private final boolean useProxy = true;
+
+  private static final ThreadPoolExecutor executor =
+      new ThreadPoolExecutor(1, 10, 1, TimeUnit.SECONDS, new LinkedBlockingQueue<Runnable>());
+  private static final List<Integer> windows = new ArrayList<Integer>();
+
+  @BeforeClass
+  public static void setUp() {
+    HandlerSettings.autoWindowOn(true);
+    logger.setLevel(Level.FINEST);
+    logHandler = new Handler() {
+
+      @Override
+      public void publish(LogRecord record) {
+        String message = record.getMessage();
```

Comment:
Yeah, I've been doing this since the beginning. Before I was taking the arguments to log call but now need to parse the string since we switched to String.format(). Are you saying a log handler shouldn't be used at all?